tests/test_job.py

```diff
@@ -943,7 +972,95 @@ def test_dependents_are_met_if_dependency_is_deleted(self):
 
         assert dependent_job.dependencies_are_met()
         assert dependent_job.get_status() == JobStatus.QUEUED
-    
+
+    def test_dependencies_are_met_at_execution_time(self):
+        queue = Queue(connection=self.testconn)
+
+        queue.enqueue_call(fixtures.say_hello, job_id="A")
+        queue.enqueue_call(fixtures.say_hello, job_id="B")
+        job_C = queue.enqueue_call(fixtures.check_dependencies_are_met, job_id="C", depends_on=["A", "B"])
+
+        w = Worker([queue])
+        w.work(burst=True)
+
+        assert job_C.result
+
+    def test_rpush_fixture(self):
+        fixtures.rpush('foo', 'bar')
+        assert self.testconn.lrange('foo', 0, 0)[0].decode() == 'bar'
+
+    def test_start_worker_fixture(self):
+        queue = Queue(name='testing', connection=self.testconn)
+        queue.enqueue(fixtures.say_hello)
+        conn_kwargs = self.testconn.connection_pool.connection_kwargs
+        fixtures.start_worker(queue.name, conn_kwargs, 'w1', True)
+        assert not queue.jobs
+
+    def test_execution_order_with_sole_dependency(self):
+        queue = Queue(connection=self.testconn)
+        key = 'test_job:job_order'
+
+        # When there are no dependencies, the two fast jobs ("A" and "B") run in the order enqueued.
+        # Worker 1 will be busy with the slow job, so worker 2 will complete both fast jobs.
+        queue.enqueue_call(fixtures.long_running_job, kwargs={'timeout': 0.5}, job_id='slow_job')
+        queue.enqueue_call(fixtures.rpush, args=[key, "A", True])
+        queue.enqueue_call(fixtures.rpush, args=[key, "B", True])
+
+        self.testconn.delete(key)
+        fixtures.burst_two_workers(queue)
+        assert queue.count == 0
+
+        jobs_completed = [v.decode() for v in self.testconn.lrange(key, 0, 1)]
+        # Job A completed first, and was done by worker 2.
+        assert jobs_completed[0] == "A:w2"
+
+        # When job "A" depends on the slow job, then job "B" finishes before "A".
+        queue.enqueue_call(fixtures.long_running_job, kwargs={'timeout': 0.5}, job_id='slow_job')
+        queue.enqueue_call(fixtures.rpush, args=[key, "A", True], depends_on='slow_job')
+        queue.enqueue_call(fixtures.rpush, args=[key, "B", True])
+
+        self.testconn.delete(key)
```

Comment:
Nitpick: mind moving all `self.testconn.delete(key)` lines to after assertions are completed? (line 1015) This provides a better separation between separate test cases.